### ISSUE TITLE
run_ner lines 154-164. Why '30'?

### ISSUE THREAD

**Issue by ypapanik** [dmis-lab/biobert#32]
Hi, congratulations for this amazing work!
I am playing with the NER code you provided and I am trying to understand the lines 154-164. If I remove this block the code fails (in detokenizer), we basically miss some words. So why do we need to split a sentence in 30-word-long sub-sentences? Is it some requirement by BERT? I also use max_seq_length 128 so it does not make sense to me. Thanks!

**Comment by jhyuklee** [comment]
@wonjininfo Any idea for this issue?

**Comment by cpmss521** [comment]
it bewildered me too,

**Comment by wonjininfo** [comment]
Hi,
Sorry for the delay. I have talked with one of our co-authors and found out the reason for the number "30". 
We wanted to prevent BioBERT from splitting a sentence within a word. In other words, we don't want subwords to be a start and end of some sequences. 

Since subwords (word piece token) count towards the max_seq_length restriction, BioBERT model will split a sentence into sequences at the length of 128 tokens regardless of where the words end at.
Therefore, a sentence is highly likely to be split incorrectly. 
We separated sentences at 30 words preemptively to prevent this. We decided the number 30 statistically (In most case, a sequence of 30 words will be transformed into a sequence with less than 128 subword tokens).
I believe this part of the NER code needs to be more sophisticated. I will update the code when it is ready (Unfortunately, it may take some time).
Thanks.
